### DISCUSSION TITLE
WebSocket?

### DISCUSSION THREAD

**Discussion by touhidurrr** [hopfenspace/runciv#3] [General]
Have you any plans to implement WebSocket?

**Comment by CrsiX** [comment]
Yes. The game data and chat workflows are based on them. However, WS are integrated into a proper REST API providing some other useful endpoints like user handling, authentication and a possible web frontend (because, why not?).

**Comment by myOmikron** [comment]
We started to define the websocket API, currently the definition is only available in [source](https://github.com/hopfenspace/runciv/blob/9bf90adda71683784eaca28e66689377fb2976c2/src/chan/ws_manager_chan.rs#L42-L92)

**Comment by touhidurrr** [comment]
kindly make the parameters camel case.
also all rest api should be available via websocket also.
i am actually not sure why do we need rest apis in the first place if we have websocket.
